src/modules/main/PlayerView.tsx

```diff
@@ -1,23 +1,159 @@
-import { Box, Typography } from '@mui/material';
+import { useEffect, useState } from 'react';
 
-import { useLocalContext } from '@graasp/apps-query-client';
+import {
+  Alert,
+  Box,
+  Button,
+  CircularProgress,
+  Container,
+  Stack,
+  Typography,
+} from '@mui/material';
 
-import { hooks } from '@/config/queryClient';
+import isEqual from 'lodash.isequal';
+
+import { AnsweredLabel, Label, Settings, SettingsKeys } from '@/@types';
+import { ANSWER_SUBMISSION_TYPE } from '@/config/constants';
+import { useAppTranslation } from '@/config/i18n';
+import { hooks, mutations } from '@/config/queryClient';
 import { PLAYER_VIEW_CY } from '@/config/selectors';
+import { APP } from '@/langs/constants';
+
+import PlayerFrame from '../common/PlayerFrame';
+
+type SubmittedAnswer = {
+  expectedId: string;
+  actualId?: string;
+};
 
 const PlayerView = (): JSX.Element => {
-  const { permission } = useLocalContext();
   const { data: appContext } = hooks.useAppContext();
-  const members = appContext?.members;
+  const { t } = useAppTranslation();
+  const { data: appData } = hooks.useAppData<{ answers: SubmittedAnswer[] }>();
+  const { data: appSettings, isLoading } = hooks.useAppSettings<Settings>({
+    name: SettingsKeys.SettingsData,
+  });
+  const { mutate: saveAppData } = mutations.usePostAppData();
+  const [answeredLabels, setAnsweredLabels] = useState<AnsweredLabel[]>([]);
+  // labels will be null only before setting the state as we cannot render all labels within container if not settled yet
+  const [labels, setLabels] = useState<null | Label[]>(null);
+
+  const { data: image } = hooks.useAppSettings({
+    name: SettingsKeys.File,
+  });
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  const answersAppData = appData?.filter(
+    ({ type }) => type === ANSWER_SUBMISSION_TYPE,
+  );
+  const settingLabels = appSettings?.[0].data.labels;
+  const lastVersion = answersAppData?.[answersAppData.length - 1];
+  const answers = lastVersion?.data.answers;
+
+  const retry = (): void => {
+    if (settingLabels) {
+      const answered = settingLabels?.map((label) => ({
+        expected: label,
+        actual: null,
+      }));
+
+      setAnsweredLabels(answered);
+      setLabels(settingLabels);
+    }
+  };
+
+  const save = (): void => {
+    const submittedAnswers = answeredLabels.map(({ expected, actual }) => ({
+      expectedId: expected.id,
+      actualId: actual?.id,
+    }));
+    saveAppData({
+      data: { answers: submittedAnswers },
+      type: ANSWER_SUBMISSION_TYPE,
+    });
+  };
+
+  useEffect(() => {
+    if (settingLabels) {
```

Comment:
I would use negative condition here, to remove one nested level. (`if (!settingLabels) { return; }`)